kratos/processes/vtk_output_process.cpp

```diff
@@ -0,0 +1,734 @@
+            for (ModelPart::ElementIterator elem_i = model_part.ElementsBegin(); elem_i != model_part.ElementsEnd(); ++elem_i)
+            {
+                if (dataCharacteristic == 1)
+                {
+                    Variable<double> elementResultVariable = KratosComponents<Variable<double>>::Get(elementResultName);
+                    double &elementResult = elem_i->GetValue(elementResultVariable);
+                    outputFile << elementResult << "\n";
+                }
+                else if (dataCharacteristic == 2)
+                {
+                    Variable<array_1d<double, 3>> elementResultVariable = KratosComponents<Variable<array_1d<double, 3>>>::Get(elementResultName);
+                    array_1d<double, 3> &elementResult = elem_i->GetValue(elementResultVariable);
+                    outputFile << elementResult[0] << " ";
+                    outputFile << elementResult[1] << " ";
+                    outputFile << elementResult[2] << "\n";
+                }
+            }
+        }
+        /*
+        outputFile << "SCALARS SPLIT_ELEMENT float 1\nLOOKUP_TABLE default\n";
+
+        // write element results for active
+        for (ModelPart::ElementIterator elem_i = model_part.ElementsBegin(); elem_i != model_part.ElementsEnd(); ++elem_i)
+        {
+            //outputFile << numberOfNodes;
+            bool is_split = elem_i->GetValue(SPLIT_ELEMENT);
+            outputFile << is_split << "\n";
+        }
+    */
+        outputFile.close();
+    }
+}
+
+//#############################################For creating vtk files in binary format##########################################################
+
+void VtkOutputProcess::WriteHeaderBinary(ModelPart &model_part)
+{
+    std::string outputFileName = GetOutputFileName(model_part);
+    std::ofstream outputFile;
+
+    outputFile.open(outputFileName, std::ios::out | std::ios::binary);
+    outputFile << "# vtk DataFile Version 4.0"
+               << "\n";
+    outputFile << "vtk output"
+               << "\n";
+    outputFile << "BINARY"
+               << "\n";
+    outputFile << "DATASET UNSTRUCTURED_GRID"
+               << "\n";
+    outputFile.close();
+}
+
+void VtkOutputProcess::WriteMeshBinary(ModelPart &model_part)
+{
+
+    WriteNodesBinary(model_part);
+
+    WriteConditionsAndElementsBinary(model_part);
+
+    WriteConditionAndElementTypesBinary(model_part);
+}
+
+void VtkOutputProcess::WriteNodesBinary(ModelPart &model_part)
+{
+    std::string outputFileName = GetOutputFileName(model_part);
+    std::ofstream outputFile;
+    outputFile.open(outputFileName, std::ios::out | std::ios::app | std::ios::binary);
+
+    // write nodes header
+    outputFile << "\nPOINTS " << model_part.NumberOfNodes() << " float"
+               << "\n";
+
+    // write nodes
+    for (ModelPart::NodeIterator node_i = model_part.NodesBegin(); node_i != model_part.NodesEnd(); ++node_i)
+    {
+        float x_coordinate = node_i->X();
```

Comment:
float?¿. Use Coordinates() to avoid calling three times the database, and use const

Reply:
Use coordinates please